client/src/components/addSourceWizard/addSourceWizard.js

```diff
@@ -43,61 +126,534 @@ class AddSourceWizard extends React.Component {
+
+    switch (sourceType) {
+      case 'network':
+        break;
+      case 'import':
+        break;
+      case 'vcenter':
+      case 'satellite':
+        if (
+          sourceNameError === '' &&
+          addressesError === '' &&
+          credentialsError === ''
+        ) {
+          stepValid = true;
+        }
+        break;
+      default:
+        stepValid = false;
+        break;
+    }
+
+    this.setState({ stepTwoValid: stepValid });
+  }
+
+  validateStepThree() {
+    const { stepOneValid, stepTwoValid } = this.state;
+
+    if (stepOneValid && stepTwoValid) {
+      this.setState({ stepThreeValid: true });
+    }
+  }
+
+  // ToDo: render and populate multiple credentials
+  renderCredentials() {}
+
+  renderStepTwo() {
+    const {
+      sourceType,
+      sourceName,
+      sourceNameError,
+      addresses,
+      addressesError,
+      credentials,
+      credentialsError,
+      fileSelect,
+      fileSelectError
+    } = this.state;
+
+    const FieldGroup = this.renderFieldGroup;
+
+    switch (sourceType) {
+      case 'network':
+        return (
+          <Form horizontal>
+            <FieldGroup
+              label={'Name'}
+              validate={sourceNameError}
+              validateMessage={sourceNameError}
+            >
+              <Form.FormControl
+                type="text"
+                value={sourceName}
+                placeholder="Enter a source name"
+                onChange={e => this.updateSourceName(e)}
+              />
+            </FieldGroup>
+
+            <FieldGroup
+              label={'Search Addresses'}
+              validate={addressesError}
+              validateMessage={addressesError}
+            >
+              <Form.FormControl
+                componentClass="textarea"
+                value={addresses}
+                rows={5}
+                placeholder="Enter an IP address:port"
+                onChange={e => this.updateAddresses(e, true)}
+              />
+              <Form.HelpBlock>
+                Comma separated, IP ranges, dns, and wildcards are valid.
+              </Form.HelpBlock>
+            </FieldGroup>
+
+            <FieldGroup
+              label={'Credentials'}
+              validate={credentialsError}
+              validateMessage={credentialsError}
+            >
+              <Form.InputGroup>
+                <Form.FormControl
+                  type="text"
+                  value={credentials}
+                  placeholder="Add a credential"
+                  onChange={e => this.updateCredentials(e, true)}
+                />
+                <Form.InputGroup.Button>
+                  <Button onClick={this.addCredential} title="Add a credential">
+                    <span className="sr-only">Add</span>
+                    <Icon type="fa" name="plus" />
+                  </Button>
+                </Form.InputGroup.Button>
+              </Form.InputGroup>
+            </FieldGroup>
+          </Form>
```

Comment:
Would be better to breakup the specifics into separate render functions and keep things like Name to the common area.